catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -157,13 +160,14 @@ public AttributeDescriptor getAttributeDescriptor(String attributeName) {
     if (attributeName == null) {
       return null;
     }
-    // TODO could this be faster?
-    for (AttributeDescriptor descriptor : descriptors) {
-      if (attributeName.equals(descriptor.getName())) {
-        return descriptor;
+
```

Comment:
✏️ this is meant to be done just once at initialization, right? Perhaps it would be best done in the constructor, or in `validateDescriptors()`. I don't have much preference as to whether it goes here or somewhere else, just sharing a thought.